src/ghga_datasteward_kit/s3_upload/uploader.py

```diff
@@ -53,28 +127,41 @@ async def schedule(self, fn: Coroutine[Any, Any, None]):
 
     async def gather(self):
         """Await all running tasks"""
-        await asyncio.gather(*self._tasks)
+        try:
+            await asyncio.gather(*self._tasks)
+        except PartUploadError:
+            # root cause should always be PartUploadError if `send_part` is scheduled as task
```

Comment:
Every exception in `send_part` is now converted into a `PartUploadError`.
It would be easier to just catch `BaseException` here instead, but not all the Information needed for the `PartUploadError` is available in this context.

This is more of a mental note to adjust the code, if the function being dispatched as task changes.

Reply:
This was the wrong way to do this either way.
Added some comments here and in `send_part` on what was actually going wrong.